packages/nx/src/generators/tree.ts

```diff
@@ -360,14 +364,20 @@ export class FsTree implements Tree {
       // TODO (v17): Remove condition
       if (gt(nxVersion, '17.0.0')) {
         throw new Error(
-          'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and ensure that Promises have resolved before the generator returns or resolves.'
+          `The tree has already been committed to disk ${
```

Comment:
File changes have already been written to disk... 

Go ahead and make this an error.